safe_transaction_service/history/management/commands/setup_service.py

```diff
@@ -231,12 +262,14 @@ def _setup_safe_master_copies(
                 },
             )
             if (
-                safe_master_copy.version != version
-                or safe_master_copy.initial_block_number != initial_block_number
+                safe_singleton_address.version != version
+                or safe_singleton_address.initial_block_number != initial_block_number
             ):
-                safe_master_copy.version = initial_block_number
-                safe_master_copy.version = version
-                safe_master_copy.save(update_fields=["initial_block_number", "version"])
+                safe_singleton_address.version = initial_block_number
```

Comment:
I guess tests were not covering this case 😉 

```suggestion
                safe_singleton_address.initial_block_number = initial_block_number
```

Reply:
Added a test to ensure that is being updated correctly.